go/pkg/casng/batching.go

```diff
@@ -665,3 +657,27 @@ func (u *BatchingUploader) UploadTree(ctx context.Context, execRoot impath.Absol
 
 	return
 }
+
+// ReplaceWorkingDir swaps remoteWorkingDir for workingDir in path which must be prefixed by root.
+// workingDir is assumed to be prefixed by root, and the returned path will be a descendent of root, but necessarily a decendent of remoteWorkingDir.
```

Comment:
did you mean "(...) not necessarily a descendent"?

also, I think a noun is descendant (with "a")

Reply:
Last I checked, both forms of the word can be used interchangeably. But yes, I missed a "not".